mcnp5/mcnp_funcs.cpp

```diff
@@ -479,45 +479,68 @@ void write_lcad_old(std::ofstream &lcadfile)
   int num_surfs = DAG->num_entities( 2 );
 
   int cmat = 0;
-  double crho, cimp = 1.0;
+  double crho;
+
+  // Detect which importances are used so all cells, including implicit
+  // complement and graveyard, have these importances
+  bool imp_n_needed = false;
+  bool imp_p_needed = false;
+  bool imp_e_needed = false;
+  for( int i = 1; i <= num_cells; ++i ) {
+    moab::EntityHandle vol = DAG->entity_by_index( 3, i );
+    if( DAG->has_prop( vol, "imp.n" )) imp_n_needed = true;
+    if( DAG->has_prop( vol, "imp.p" )) imp_p_needed = true;
+    if( DAG->has_prop( vol, "imp.e" )) imp_e_needed = true;
+  }
 
   // write the cell cards
   for( int i = 1; i <= num_cells; ++i ) {
 
     moab::EntityHandle vol = DAG->entity_by_index( 3, i );
     int cellid = DAG->id_by_index( 3, i );
-    // set default importances for p and e to negative, indicating no card should be printed.
-    double imp_n = 1, imp_p = -1, imp_e = -1;
+    // set default importances to negative, indicating no card should be printed.
+    double imp_n = -1, imp_p = -1, imp_e = -1;
 
     if( DAG->has_prop( vol, "imp.n" )) {
       get_real_prop( vol, cellid, "imp.n", imp_n );
+    } else if( imp_n_needed ) {
+      imp_n = 0;
```

Comment:
I think that was the default. If there is no graveyard I think there should be a failure if there already isn't.